scalanet/src/io/iohk/scalanet/peergroup/kademlia/KRouter.scala

```diff
@@ -23,7 +29,7 @@ class KRouter[A](
     val clock: Clock = Clock.systemUTC(),
     val uuidSource: () => UUID = () => UUID.randomUUID(),
     val rnd: Random = new SecureRandom()
-) {
+)(implicit codec: Codec[A]) {
```

Comment:
I am not sure why the krouter needs `Codec[A]` ?

Reply:
It needs the Codec[A] for use the function "verify" of NodeRecord. This function needs it to encode the "routingAddress" and "messagingAddress" parameters of NodeRecord.